app/assets/javascripts/govuk_publishing_components/lib/govspeak/youtube-link-enhancement.js

```diff
@@ -1,48 +1,78 @@
     this.$element = $element
   }
+  var playList = []
+  var players = []
+
+  window.onYouTubePlayerAPIReady = function () {
+    for (var i = 0; i < playList.length; i++) {
+      var currentPlayer = YoutubeLinkEnhancement.prototype.createVideoPlayer(playList[i])
+      players[i] = currentPlayer
+    }
+  }
 
   YoutubeLinkEnhancement.prototype.init = function () {
-    var $youtubeLinks = this.$element.find('a[href*="youtube.com"], a[href*="youtu.be"]')
+    var $youtubeLinks = this.$element[0].querySelectorAll('a[href*="youtube.com"], a[href*="youtu.be"]')
     var _this = this
-    $youtubeLinks.each(function () {
-      var $link = $(this)
-
+    _this.loadYoutubeApi()
+    for (var i = 0; i < $youtubeLinks.length; ++i) {
+      var $link = $youtubeLinks[i]
       if (_this.hasDisabledEmbed($link)) {
         return true
       }
-
-      var videoId = YoutubeLinkEnhancement.parseVideoId($link.attr('href'))
+      var videoId = YoutubeLinkEnhancement.parseVideoId($link.getAttribute('href'))
       if (videoId) {
         _this.embedVideo($link, videoId)
       }
-    })
+    }
+  }
+
+  YoutubeLinkEnhancement.prototype.loadYoutubeApi = function () {
+    var tag = document.createElement('script')
+    tag.src = 'https:///www.youtube.com/iframe_api'
+    var firstScriptTag = document.getElementsByTagName('script')[0]
+    firstScriptTag.parentNode.insertBefore(tag, firstScriptTag)
   }
 
   YoutubeLinkEnhancement.prototype.hasDisabledEmbed = function ($link) {
-    return $link.attr('data-youtube-player') === 'off'
+    return $link.getAttribute('data-youtube-player') === 'off'
   }
 
   YoutubeLinkEnhancement.prototype.embedVideo = function ($link, videoId) {
-    var $mediaPlayer = $('<span class="media-player" />')
-
-    $link.parent().replaceWith($mediaPlayer)
-
-    var protocol = this.protocol()
+    var parent = $link.parentNode
+    var youtubeVideoContainer = '<div class="youtube-video-container">' +
+    // '<span id="' + videoId + '"></span></div>'
+    '<iframe id="' + videoId + '" type="text/html" src="https://www.youtube-nocookie.com/embed/' +
+    videoId + '?enablejsapi=1" frameborder="0" allowfullscreen></iframe></div>'
+    parent.innerHTML = youtubeVideoContainer
+    playList.push({
+      id: videoId,
+      videoId: videoId,
+      title: $link.textContent
+    })
+  }
 
-    $mediaPlayer.player({
-      id: YoutubeLinkEnhancement.nextId(),
-      media: videoId,
-      captions: this.captions($link),
-      url: protocol + '//www.youtube.com/apiplayer?enablejsapi=1&version=3&playerapiid='
+  YoutubeLinkEnhancement.prototype.createVideoPlayer = function (playlistItem) {
+    return new window.YT.Player(playlistItem.id, {
+      videoId: playlistItem.videoId,
+      host: 'https://www.youtube-nocookie.com',
+      playerVars: {
+        enablejsapi: 1,
+        rel: 0, // don't show related videos
+        disablekb: 1, // insure the player does not respond to keyboard controls (WCAG)
```

Comment:
Could you link to the WCAG guideline in this comment please?

Reply:
done